### ISSUE TITLE
Unable to "Create Key" using the http API

### ISSUE THREAD

**Issue by hummerbliss** [LeSuisse/vault-gpg-plugin#3]
Hello,

I am just getting started with vault and gpg plugin and I get the following error when I try to create a key. Hope I am not missing something obvious. Vault started in server -dev mode.

```
curl --header "X-Vault-Token:$VAULT_TOKEN" --request POST --data @payload.json http://127.0.0.1:8200/v1/
vault-gpg-plugin/keys/my-key
```
and the response I get is 

```
{"errors":["1 error occurred:\n\n* internal error"]}
```

On the vault server side, I see the following messages

```
2018/01/31 03:08:07.585993 [INFO ] core: successful mount: path=vault-gpg-plugin/ type=plugin
2018/01/31 03:10:21.703524 [ERROR] core: failed to run existence check: error=gob: type not registered for interface: json.Number
2018/01/31 03:11:12.182289 [ERROR] rollback: error rolling back: path=vault-gpg-plugin/ error=unexpected EOF
```

Vault: 0.9.3
vault-gpg-plugin: 0.1.3

I was able import and enable plugin as you can see from the first log message.

**Comment by LeSuisse** [comment]
Hi,

It should have been fixed with Vault 0.9.3 and the release 0.1.3 of the plugin :/ .

I will try to reproduce the issue on my side and keep you updated. In the meantime, if you are fine with a 2048 bits GPG key you can omit the parameter `key_bits` from the payload and it should work as expected.

**Comment by hummerbliss** [comment]
Hi, thank you, yup can omit the parameter (2048 is fine for my usecase).

On a hunch (since it was complaining about Json.Number) I changed the payload to represent the key_bits from number to string and it worked, 

this was the old payload (as given in the README.md)

```
{
  "real_name": "John Doe",
  "email": "john.doe@example.com",
  "key_bits": 4096
}
```

New payload

```
{
  "real_name": "John Doe",
  "email": "john.doe@example.com",
  "key_bits": "4096"
}
```

although not sure if it did really use 4096 bits.

**Comment by LeSuisse** [comment]
Ok so I confirm the issue even with the latest releases.

This should be fixed if the plugin uses the new gRPC interface available since Vault 0.9.2 but this interface has a bug which prevent LIST operations. Since being able to list existing GPG keys is useful, I have forced the fallback to the netRPC interface in 4ea5d25a37769bc4e3f8e3e19659e0c080aef393 before releasing the 0.1.3.

A fix is planned with Vault 0.9.4, so I will release a new version of the plugin once it's available to use the gRPC interface. In the meantime I have updated the release note of the plugin and the sample payload to use a string instead of an int (yes, it works as expected when you do that).

**Comment by LeSuisse** [comment]
FYI, the list operation issue has been fixed in Vault. As soon as 0.9.4 is released I will update the plugin so it's possible to both list the keys and choose the size as expected.

**Comment by LeSuisse** [comment]
Vault 0.9.4 has been released and I released a new version of the plugin (v0.1.4).

With the combination of these latest releases both features now work without workarounds: `key_bits` can be used as an int as expected and the keys can be listed.
